src/sentry/models/groupinbox.py

```diff
@@ -75,18 +81,21 @@ class Meta:
         indexes = (models.Index(fields=("project", "date_added")),)
 
 
-def add_group_to_inbox(group, reason, reason_details=None):
-    if reason_details is not None:
-        if "until" in reason_details and reason_details["until"] is not None:
-            reason_details["until"] = reason_details["until"].replace(microsecond=0).isoformat()
+def add_group_to_inbox(
+    group: Group,
+    reason: GroupInboxReason,
+    reason_details: InboxReasonDetails | None = None,
+) -> GroupInbox:
+    if reason_details is not None and reason_details["until"] is not None:
+        reason_details["until"] = reason_details["until"].replace(microsecond=0)
```

Comment:
If I keep `isoformat()` I get this error:
> src/sentry/models/groupinbox.py:90: error: Value of "until" has incompatible type "str"; expected "datetime | None"  [typeddict-item]
<img width="529" alt="image" src="https://github.com/user-attachments/assets/5be57c0e-dc64-40dd-849c-6cbc9ce2ebf3" />